include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -1198,8 +1207,15 @@ namespace internal
                   dof_info[no].cell_active_fe_index[counter] = fe_index;
                 local_dof_indices.resize(dof_info[no].dofs_per_cell[fe_index]);
                 cell_it->get_dof_indices(local_dof_indices);
-                dof_info[no].read_dof_indices(local_dof_indices,
-                                              lexicographic[no][fe_index],
+
+                local_dof_indices_plain.resize(local_dof_indices.size());
+                for (unsigned int i = 0; i < local_dof_indices.size(); ++i)
+                  local_dof_indices_plain[i] =
+                    local_dof_indices[lexicographic[no][fe_index][i]];
+
+                dof_info[no].read_dof_indices(local_dof_indices_plain,
```

Comment:
Between these two lines, the hanging-node algorithm will be added. See: 

https://github.com/dealii/dealii/blob/4f01070cdbd527fc9c41479a7bd3938f85c7ded9/include/deal.II/matrix_free/matrix_free.templates.h#L1227-L1246